pkg/kvstore/allocator/allocator.go

```diff
@@ -699,11 +699,16 @@ func (a *Allocator) RunGC() error {
 }
 
 func (a *Allocator) recreateMasterKey(id idpool.ID, value string, reliablyMissing bool) {
+	createOP := kvstore.UpdateIfDifferent
+	if reliablyMissing {
+		createOP = kvstore.CreateOnly
+	}
+
 	keyPath := path.Join(a.idPrefix, id.String())
 
-	// Use of CreateOnly() ensures that any existing potentially
-	// conflicting key is never overwritten.
-	err := kvstore.CreateOnly(context.TODO(), keyPath, []byte(value), false)
+	// Use of UpdateIfDifferent() and CreateOnly() ensures that any
+	// existing potentially conflicting key is never overwritten.
+	err := createOP(context.TODO(), keyPath, []byte(value), false)
```

Comment:
Would UpdateIfDifferent() explicitly update a conflicting key (or value?), if the old one is different? Or maybe I don't understand what "conflicting key" means?

Reply:
Fixed